### PR TITLE
Use "ID" instead of "name" for tables in new RPCs

### PR DESCRIPTION
**Pull request by ctubbsii**
Use TableId and NamespaceId instead of tableName and namespaceName in
new RPCs for 2.1

* Standardize naming conventions for table and namespace IDs and names
  in thrift IDL files
* Create Tid ("thrift ID") wrapper for strings, to avoid string types
  directly in thrift IDL
* Change client code where necessary to pass ID types instead of string
  names, by looking up the ID right away, and using that in the RPC
  methods, and change corresponding server code to eliminate unnecessary
  ID lookups
* Eliminate redundant name validation in client code (it's done in the
  ID lookup)

### CODE REVIEW

File: core/src/main/thrift-gen-java/org/apache/accumulo/core/clientImpl/thrift/TDiskUsage.java [hunk 1]
```diff
@@ -28,18 +28,18 @@
...
   private static final org.apache.thrift.scheme.SchemeFactory TUPLE_SCHEME_FACTORY = new TDiskUsageTupleSchemeFactory();
 
-  public @org.apache.thrift.annotation.Nullable java.util.List<java.lang.String> tables; // required
+  public @org.apache.thrift.annotation.Nullable java.util.List<java.lang.String> tableNames; // required
   public long usage; // required
 
   /** The set of fields this struct contains, along with convenience methods for finding and manipulating them. */
   public enum _Fields implements org.apache.thrift.TFieldIdEnum {
-    TABLES((short)1, "tables"),
+    TABLE_NAMES((short)1, "tableNames"),
```
Comment (milleruntime) [inline core/src/main/thrift-gen-java/org/apache/accumulo/core/clientImpl/thrift/TDiskUsage.java:42]: Will this cause issues with older RPC calls? Maybe that's OK to eliminate the ambiguity of "table" since to make it backwards compatible we would have to keep it around.

File: core/src/main/thrift/client.thrift [hunk 2]
```diff
@@ -22,6 +22,11 @@ namespace cpp org.apache.accumulo.core.clientImpl.thrift
 include "security.thrift"
 include "trace.thrift"
 
+// Thrift Id, a wrapper for AbstractId over thrift
+struct Tid {
```
Comment (milleruntime) [inline core/src/main/thrift/client.thrift:26]: What about `TAbstractId`? There are places in the code where we use "tid" for transactionId.

Comment (dlmarion) [reply core/src/main/thrift/client.thrift:26]: or `idType`

**Review by dlmarion — APPROVED** [review]

File: core/src/main/thrift-gen-java/org/apache/accumulo/core/clientImpl/thrift/TDiskUsage.java (hunk 1 above)
Comment (ctubbsii) [reply core/src/main/thrift-gen-java/org/apache/accumulo/core/clientImpl/thrift/TDiskUsage.java:42]: I spent some time reading various guides on Thrift RPC compatibility. From everything I can tell, the RPC compatibility depends only on the data type (which I only changed in stuff that's new in 2.1) and the numeric field identifier (which I didn't change) to match up the fields for deserialization. Field names don't matter. However, RPC method names (which I didn't change) do matter.

What is broken is class path dependencies... which is fine... because you shouldn't be mixing and matching Accumulo jars from different versions, and the Thrift IDL is not public API for users, so they shouldn't have any references to these in their code anyway.

That said, I'm pretty sure we've already broken client-server RPC compatibility between 2.0 and 2.1, so it is not a problem to introduce additional changes that change the RPC layer a bit. But, I avoided making any such changes here, before taking a closer to look to ensure compatibility was already broken and there would be no additional harm.

File: core/src/main/thrift/client.thrift (hunk 2 above)
Comment (ctubbsii) [reply core/src/main/thrift/client.thrift:26]: I wasn't sure if I should even keep this type at all, or if I should make two separate types, TTableId and TNamespaceId, which corresponding utility methods, to further disambiguate. I was also considering TTable and TNamespace types that would have an ID/Name pair, to make these useful to carry name information for log messages so that way we don't have to do a name lookup on the server side for a log message, and so we could see the name the client thought the table was at the time it did the ID lookup, in case the table gets renamed before the log message is printed.

In short, there's room to grow here, but I'm not 100% sure which is the best direction.

Comment (Manno15) [reply core/src/main/thrift/client.thrift:26]: I am in favor of the TTable/TNamespace idea that contains a pair of ID/Name.

**Comment by ctubbsii** [comment]
I'm going to take another pass at this, using `TTable` and `TNamespace` instead.
